eng/versioning/external_dependencies.txt

```diff
@@ -0,0 +1,200 @@
+# effectively means that each dependency needs to be listed here.
+org.junit.jupiter:junit-jupiter;5.4.2
+org.junit.jupiter:junit-jupiter-api;5.4.2
+org.junit.jupiter:junit-jupiter-engine;5.4.2
+org.junit.jupiter:junit-jupiter-migrationsupport;5.4.2
+org.junit.jupiter:junit-jupiter-params;5.4.2
+org.junit.platform:junit-platform-commons;1.4.2
+org.junit.platform:junit-platform-console;1.4.2
+org.junit.platform:junit-platform-engine;1.4.2
+org.junit.platform:junit-platform-launcher;1.4.2
+org.junit.platform:junit-platform-reporting;1.4.2
+org.junit.platform:junit-platform-runner;1.4.2
+org.junit.platform:junit-platform-suite-api;1.4.2
+org.junit.platform:junit-platform-testkit;1.4.2
+org.junit.vintage:junit-vintage-engine;5.4.2
+org.openjdk.jmh:jmh-core;1.22
+org.openjdk.jmh:jmh-generator-annprocess;1.22
+org.spockframework:spock-core;1.3-groovy-2.5
+org.testng:testng;6.14.3
+uk.org.lidalia:slf4j-test;1.2.0
+
+## Maven Tools versions
+com.azure:sdk-build-tools;1.0.0
+com.beust:jcommander;1.58
+com.google.code.findbugs:jsr305;3.0.2
+com.nimbusds:nimbus-jose-jwt;6.0.1
+com.nimbusds:oauth2-oidc-sdk;6.14
+com.puppycrawl.tools:checkstyle;8.24
+commons-io:commons-io;2.5
+commons-validator:commons-validator;1.6
+io.dropwizard.metrics:metrics-core;4.0.5
+io.dropwizard.metrics:metrics-graphite;4.0.5
+io.dropwizard.metrics:metrics-jvm;4.0.5
+io.reactivex.rxjava2:rxjava;2.2.4
+net.java.dev.jna:jna-platform;5.4.0
+org.apache.commons:commons-collections4;4.2
+org.apache.commons:commons-lang3;3.8.1
+org.apache.commons:commons-text;1.6
+org.apache.maven.plugins:maven-antrun-plugin;1.8
+org.apache.maven.plugins:maven-assembly-plugin;2.2
+org.apache.maven.plugins:maven-checkstyle-plugin;3.1.0
+org.apache.maven.plugins:maven-clean-plugin;3.0.0
+org.apache.maven.plugins:maven-compiler-plugin;3.8.1
+org.apache.maven.plugins:maven-eclipse-plugin;2.8
+org.apache.maven.plugins:maven-failsafe-plugin;2.22.0
+org.apache.maven.plugins:maven-help-plugin;2.1.1
+org.apache.maven.plugins:maven-jar-plugin;3.1.2
+org.apache.maven.plugins:maven-javadoc-plugin;3.1.1
+org.apache.maven.plugins:maven-jxr-plugin;3.0.0
+org.apache.maven.plugins:maven-project-info-reports-plugin;3.0.0
+org.apache.maven.plugins:maven-release-plugin;2.5.3
+org.apache.maven.plugins:maven-resources-plugin;2.4.3
+org.apache.maven.plugins:maven-site-plugin;3.7.1
+org.apache.maven.plugins:maven-source-plugin;3.0.1
+org.apache.maven.plugins:maven-surefire-plugin;3.0.0-M3
+org.apidesign.javadoc:codesnippet-doclet;0.32
+org.codehaus.mojo:build-helper-maven-plugin;3.0.0
+org.codehaus.mojo:exec-maven-plugin;1.2.1
+org.codehaus.mojo:properties-maven-plugin;1.0.0
+org.codehaus.mojo:xml-maven-plugin;1.0
+org.eclipse.jetty:jetty-maven-plugin;9.3.22.v20171030
+org.eclipse.m2e:lifecycle-mapping;1.0.0
+org.jacoco:jacoco-maven-plugin;0.8.4
+org.jacoco:org.jacoco.agent;0.8.4
+org.mockito:mockito-core;3.0.0
+org.nanohttpd:nanohttpd;2.3.1
+
+# External Dependency Exceptions
```

Comment:
It's worth noting that these dependency exceptions, for those versions that are lower, are going to have written up issues for the specific area to look at updating to the common version. For those versions that are higher issues will be written up for us to investigate moving the common version to the higher version.